### PR TITLE
User filters are now in place

### PR DESCRIPTION
**Pull request by kerrd89**
The default is to show the current week, with Spikes from the last 24 hours being showed so on Friday we can still see the spikes.  

If a user selects another date, it shows the same range from that date.

We can get more specific with this so if a user selects a date it will explicitly show that date, but I would look to you all for feedback on whether we want to do that or not.

### CODE REVIEW

File: lib/components/Spikes.js [hunk 1]
```diff
@@ -12,12 +12,16 @@ export default class Spikes extends Component {
...
+    this.setState({ dateFilter: Date.now() });
+  }
+
   filterSpikes(spikes) {
     const { updateCount, user } = this.props;
     const { dateFilter } = this.state
     return map(spikes, (spike) => {
       if(spike.appr) {
-        if(dateFilter === 'all' || dateFilter === spike.spikeDate) {
+          if(spike.spikeDate > dateFilter - 86400000  && spike.spikeDate < dateFilter + 604800000) {
```
Comment (swanie21) [inline lib/components/Spikes.js]: @kerrd89 Just curious what do those unix time stamps represent? What are you calculating from the current  time of Date.now()?

File: lib/components/Spikes.js [hunk 2]
```diff
@@ -1,27 +1,63 @@
...
-  return (
-    <section>
-      <h1 className="SpikesTitle">Student-Led Spikes</h1>
-      <section className="AllSpikes">{allSpikes}</section>
-    </section>
-  )
-}
 
-export default Spikes;
+export default class Spikes extends Component {
```
Comment (swanie21) [inline lib/components/Spikes.js]: I thought this component was going to have to turn stateful at some point...looks like now was the turning point.

**Comment by Peter-Springer** [comment]
Can you see the admin page on this branch? On the current iteration of master I can not see the admins view even though I am on the admin list. Just trying to track down if this is from me changing db rules or something else.

**Comment by Peter-Springer** [comment]
never mind got it

**Pull request merged**